docs/markdown/Contributing.md

```diff
@@ -170,6 +170,13 @@ Various constructs containing `?` are used to indicate platform specific
 filename variations (e.g. `?so` represents the platform appropriate suffix for a
 shared library)
 
+* `fails_on.txt` specifies the stage, where a failing tests is expected to fail.
+If this file is present, the test case must fail specified stage. Possible stages
+are:
+  - meson
+  - build
+  - test
+
```

Comment:
It might be worthwhile to make this slightly more generic (e.g. a key=value in dict in a json file or whatever), so we don't have to add yet another file for yet another test configuration variation in future.

Or maybe you could just put this test in 'test cases/failing'? (since no validation is currently done of how those failing tests are failing (so failing due to missing cmake is indistinguishable))

Reply:
I am planning to refactor the testing framework (for the next big release probably), so that there is only one big test definition file and we don't have `installed_files.txt`, `test_args.txt`, `setup_env.json`, `test_matrix.json` and `fails_on.txt`. So, having one extra file now is only a bit more work for me later on.

I can't put this in the failing directories, since it fails on build, and not on configure.

Reply:
You can do something like:

```meson
if not find_program('cmake': required: false).found()
  executable('bob', 'a_non_compiling_source.cpp')
endif
```

Reply:
Moved it to `failing build` for now.